src/sql/src/plan.rs

```diff
@@ -1547,7 +1548,17 @@ pub enum ExecuteTimeout {
 pub enum IndexOption {
     /// Configures the logical compaction window for an index. `None` disables
     /// logical compaction entirely.
-    LogicalCompactionWindow(Option<Duration>),
+    LogicalCompactionWindow(Option<mz_repr::Timestamp>),
+}
+
+#[derive(Clone, Debug)]
+pub enum CompactionWindow {
+    /// Unspecified by the user, use a system-provided default.
+    Default,
+    /// Disable logical compaction entirely.
+    Disabled,
```

Comment:
Is this meant to be constructible at the moment? It doesn't seem to be! Is the idea that we'll introduce `RETAIN HISTORY = FOREVER` in the future?

Reply:
It is written this way to have compatibility with index options which allow for this: https://github.com/MaterializeInc/materialize/blob/a75510b3786759a7e438c1e0f860922f4603002b/src/sql/src/plan.rs#L1548

In a later PR I would like to refactor the index planning and controller read policy code to use this new enum instead of passing around `Option<milliseconds or Timestamp or Duration>`.

Reply:
The principal motivation for this struct is to describe exactly what a `None` Option means. There are some places it means "I want NO user compaction performed" and others it means "no one specified anything, use the default". With the current scheme we'd have some `Option<Option<Duration>>` in a few places. I thought this would improve that greatly, although the refactor is getting annoying.